include/envoy/upstream/cluster_manager.h

```diff
@@ -350,6 +358,26 @@ class ClusterManagerFactory {
    * Returns the secret manager.
    */
   virtual Secret::SecretManager& secretManager() PURE;
+
+  /**
+   * Returns a struct with all the Stats::StatName objects needed by
+   * ClusterManager. This helps factor out some relatively heavy name
+   * construction which occur when there is a large CDS update during operation,
+   * relative to recreating all stats from strings on-the-fly.
```

Comment:
Wouldn't these be Cluster/upstream stats and not cluster manager stats which I think would be global and created once?

Reply:
yes, will rename.

Reply:
I reverted the ClusterManager stats to their previous state.